17_differential_abundance.Rmd

```diff
@@ -6,20 +6,411 @@ chapterPreamble()
+whether taxa are differentially abundant can be answered by looking at the
+`res` object, which now contains dataframes with the coefficients, 
+standard errors, p-values and q-values. Conveniently, there is a dataframe
+`diff_abn`. Here, for each taxon it is indicated whether it is differentially
+abundant between the groups. Below we show the first 6 entries of this
+dataframe:  
+
+```{r}
+kable(head(res$diff_abn))
+```
+
+
+
+### Comparison of the methods
+
+When we compare the methods in the context of a research question, we could
+look at e.g. at whether they agree based on the applied decision criterion
+(e.g. adjusted p value < 0.05).  That is what we illustrate here. First let us
+print how many taxa were identified by each method to begin:
+
+
+```{r}
+# how many taxa are significant for wilcoxon test
+sum(wilcoxon_p$p_adjusted < 0.05, na.rm = TRUE)
+# DESeq2 
+sum(df$padj<0.05, na.rm = TRUE)
+# and ANCOM-BC
+sum(out$res$diff_abn$phenoObese)
+```
+
+
+
+In the next step we will look at the intersection of identified taxa. Here we
+encountered a common issue that we want to warn you about regarding the names
+of taxa.  R might change these names, e.g. if you generate a dataframe while
+the given names are not suitable to be column names. This happened here with
+`Escherichia/Shigella`, which was changed to `Escherichia.Shigella`
+automatically by R in the Wilcoxon test section. That caused some of the `NA`
+values where we expected p values (DESeq2 and Wilcoxon output) or boolean
+values (ANCOMBC output). Make sure to always inspect why you do not get the
+output you would have expected. Since `Escherichia/Shigella` was FALSE
+(adjusted p > 0.05) for all methods, I could easily work around this below by
+setting the `NA` to FALSE.
+
+
+```{r}
+# create a dataframe that indicates for each genus and method if it was DA
+summ <- full_join(
+  rownames_to_column(df, "genus"),
+  rownames_to_column(out$res$diff_abn, "genus"),
+  by = "genus") %>%
+  full_join(wilcoxon_p, by = "genus") %>%
+  # rename according to which padj value belong to which method
+  select(genus, wilcoxon = p_adjusted, deseq2 = padj, ancombc = phenoObese) %>%
+  # apply decision criterion of padj <= 0.05; address issue based on naming 
+  # error (see description above)
+  mutate(
+    across(c(deseq2, wilcoxon), function(x) x <= 0.05),
+    across(c(deseq2, wilcoxon, ancombc), function(x) ifelse(is.na(x), FALSE, x)),
+    score = rowSums(across(c(deseq2, ancombc, wilcoxon)))
+  )
+
+# which genera are identified by all methods?
+filter(summ, score == 3) %>% kable()
+```
+
+We see that while each method at least identified 14 genera as differentially
+abundant, they only agree on 5 genera. Lets plot those 5 genera:
+
+
+```{r}
+# create a plot for each genus where the score is indicated in the title
+plots <- pmap(select(summ, genus, score), function(genus, score) {
+  ggplot(daa, aes_string("x", genus)) +
+  geom_boxplot() +
```

Comment:
How about adding `geom_jitter(width=0.2, alpha=0.5)` or similar? This can potentially uncover interesting differences that some methods can/not take into account.